kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -474,4 +543,103 @@ private void configurationEnforcementDisable() throws DBusException {
         }
         this.configurationEnforcementHandlerIsArmed = false;
     }
+
+    private String getDeviceIdFromNM(String devicePath) throws DBusException {
+        Properties nmModemProperties = this.dbusConnection.getRemoteObject(NM_BUS_NAME, devicePath,
+                Properties.class);
+        String deviceId = (String) nmModemProperties.Get(NM_DEVICE_BUS_NAME + ".Modem", "DeviceId");
+        logger.debug("Found DeviceId {} for device {}", deviceId, devicePath);
+        return deviceId;
+    }
+
+    private Map<DBusPath, Map<String, Map<String, Variant<?>>>> getManagedObjects() throws DBusException {
+        ObjectManager objectManager = this.dbusConnection.getRemoteObject(MM_BUS_NAME,
+                MM_BUS_PATH, ObjectManager.class);
+        Map<DBusPath, Map<String, Map<String, Variant<?>>>> managedObjects = objectManager.GetManagedObjects();
+        logger.debug("Found Managed Objects {}", managedObjects.keySet());
+        return managedObjects;
+    }
+
+    private Optional<String> getModemPathFromMM(Map<DBusPath, Map<String, Map<String, Variant<?>>>> managedObjects,
+            String deviceId) {
+        Optional<String> modemPath = Optional.empty();
+        Optional<Entry<DBusPath, Map<String, Map<String, Variant<?>>>>> modemEntry = managedObjects.entrySet()
+                .stream().filter(entry -> {
+                    String modemDeviceId = (String) entry.getValue().get(MM_MODEM_NAME)
+                            .get("DeviceIdentifier")
+                            .getValue();
+                    return modemDeviceId.equals(deviceId);
+                }).findFirst();
+        if (modemEntry.isPresent()) {
+            modemPath = Optional.of(modemEntry.get().getKey().getPath());
+        }
+        return modemPath;
+    }
+
+    private Optional<String> getModemPath(String devicePath) throws DBusException {
```

Comment:
Should we add some form of convention to distinguish methods that are called for MM or NM?

NM has a modem type and a modem path too... From the method name alone I don't know what I'm calling.

Reply:
That's true. However, this method involves both NM and MM. I can call it `getMMModemPath`, since there is a `getModemPathFromMM`...